src/sqlfluff/cli/commands.py

```diff
@@ -383,6 +422,35 @@ def dialects(**kwargs) -> None:
     click.echo(format_dialects(dialect_readout), color=c.get("color"))
 
 
+@cli.command()
+@common_options
+@click.option(
+    "--save-path",
+    default=None,
+    type=click.Path(),
+    help="Where to save the completion script. Defaults to ~/.config/sqlfluff",
+)
+@click.argument(
+    "shell_type",
+    default="bash",
+    type=click.Choice(["bash", "zsh", "fish"], case_sensitive=False),
+    # help="What shell to generate autocompletion script for.",
+)
+def autocomplete(shell_type: str, save_path: str, **kwargs: Any) -> None:
+    """Generate autocompletion script.
+
+    This saves an autocompletion shell script to ~/.config/sqlfluff that
+    the user can enable by adding the following line to their shell config:
+    . ~/.config/sqlfluff/sqlfluff-complete.{shell_type}
+
+    N.B. this is only available bash, zsh, and fish.
+    """
+    config = get_config(**kwargs)
+    autocomplete_dict = generate_autocomplete_script(shell_type, save_path)
+    click.echo(format_autocomplete(autocomplete_dict), color=config.get("color"))
+    _completion_message(config)
```

Comment:
Where is this function defined?

Reply:
`_completion_message`?
It's just the usual `All Finished 📜 🎉!` message: https://github.com/sqlfluff/sqlfluff/blob/68f479a53cfcabe67379dfb96539e516bf80d3c8/src/sqlfluff/cli/commands.py#L744